Adding print functionality

lomamech opened the pull request:
Adding print functionality

Update bhBreadcrumb.js component for adding property print
Update template for breadCrumb with the using of bh-breadcrumb
Implement patient invoice report in PDF
Update inventory list, voucher registry for Prevent modale windows
Adding route for manage invoice / patient / report

@jniles

lomamech committed:
Resolve conflict with branch Master

jniles commented on server/controllers/finance/patientInvoice.js:
```diff
@@ -290,3 +299,45 @@ function reference(req, res, next) {
+    project : req.session.project
+  };
+
+  let invoiceListQuery =
```
This code is the exact same code as [the patientInvoice list() function](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/patientInvoice.js#L53).  Since it is a duplication, I propose that you use the same function.

I propose that you make a new function in the `patientInvoice` file called `listInvoices()`.  You could then write:

``` js
// GET /invoices
function list(req, res, next) {
  listInvoices()
  .then(function (invoices) {
    res.status(200).json(invoices);
  })
  .catch(next)
  .done();
}

// GET /invoices/patient/report
function getPatientInvoice(req, res, next) {

  const request = { /* some values */ };

  listInvoices()
  .then(invoices => listReceipt.build(invoices, request):
   // ... etc ...
  })
  .catch(next)
  .done();
}
```

Remember, in software engineering, we [don't like to repeat ourselves](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

jniles commented on server/controllers/finance/receipts/list.handlebars:
```diff
@@ -0,0 +1,54 @@
+                <td>{{ reference }}</td>
+                <td>{{ date date }}</td>
+                <td>{{ patientNames }}</td>
+                <td>{{ currency cost }}</td>
```
The currency filter should take in a currency_id.  Like this:

``` hbs
<td>{{ currency cost currency_id }}</td>
```

What currency should it?

jniles commented on server/controllers/finance/receipts/list.js:
```diff
@@ -0,0 +1,42 @@
+'use strict';
```
This file is actually a report of all invoices, correct?   I would put it in `/finance/reports/invoices` to avoid developer confusion.

The same can be done with the `.handlebars` file.

jniles commented on server/controllers/finance/receipts/list.js:
```diff
@@ -0,0 +1,42 @@
+
+
+  let queryString  = request.query;
+  let renderTarget = (queryString && queryString.renderer) ? queryString.renderer : defaultRender;
```
Just a comment (no need to change this line of code), but you could do the same thing like this:

``` js
const renderTarget = queryString && queryString.renderer || defaultRender;
```

Pretty cool!

lomamech committed:
Using the function listInvoices in controllers/finance/patientInvoice.js

lomamech committed:
Put currency_id in filter for the currency

lomamech committed:
Sanitaze the reports invoices
Changement of repository for list and list.hanflerbars
Remove the unused repository receipts

lomamech committed:
Sanitaze code

lomamech committed:
Fix the size of button

lomamech committed:
Update the orientation remplace landscape by portrait

lomamech committed:
Resolve conflict in translate file
